### ISSUE TITLE
leading and trailing periods / dots not tokenized as part of identifier

### ISSUE THREAD

**Issue by kevinushey** [REditorSupport/vscode-R#1183]
Given an R document with the code:

```
.hello.world.
```

The leading + trailing `.`s are tokenized as separate entities.

<img width="406" alt="Screen Shot 2022-08-25 at 4 37 25 PM" src="https://user-images.githubusercontent.com/1976582/186787387-5ed5ef39-4bc8-467d-80c5-0f317a889fbd.png">

I believe the use of `\b` here forces VSCode to consider `.` as a word boundary, and so it fails to tokenize the whole identifier as a single token:

https://github.com/REditorSupport/vscode-R/blob/da579cc17a9485f7a11bf5e75bf6b9c347eca116/syntax/r.json#L165-L168

If so, I think those `\b`s can be safely removed?

**Comment by renkun-ken** [comment]
Yes, I think you are right. We should remove those `\b`. Do you find unnecessary `\b` in other matches?

**Comment by kevinushey** [comment]
I guess another example `42.`, which R parses as a single number but appears to be tokenized as `42` and `.`:

<img width="474" alt="Screen Shot 2022-08-25 at 5 33 20 PM" src="https://user-images.githubusercontent.com/1976582/186792032-d821bdc8-34ea-4218-a1be-5c7618f372a1.png">

https://github.com/REditorSupport/vscode-R/blob/da579cc17a9485f7a11bf5e75bf6b9c347eca116/syntax/r.json#L123-L126

And also for references to packages containing a `.` in their name:

<img width="214" alt="Screen Shot 2022-08-25 at 5 39 10 PM" src="https://user-images.githubusercontent.com/1976582/186792461-ad0d5148-3c06-4303-ad1f-dad860487f57.png">

I'll poke around a bit more for other edge cases...

**Comment by gowerc** [comment]
Could one potential edge case be on S3 methods ? i.e.

as.data.frame.list() 

though tbh I'm not exactly sure what you would want to tokenise here :s

**Comment by eitsupi** [comment]
unstale
